plugin/storage/es/spanstore/reader.go

```diff
@@ -337,7 +337,17 @@ func (s *SpanReader) multiRead(ctx context.Context, traceIDs []model.TraceID, st
 		}
 		searchRequests := make([]*elastic.SearchRequest, len(traceIDs))
 		for i, traceID := range traceIDs {
-			query := elastic.NewTermQuery("traceID", traceID.String())
+			// TODO remove in newer versions, added in Jaeger 1.16
+			// read ID without leading zeros
+			// https://github.com/jaegertracing/jaeger/pull/1956 added leading zeros to IDs
+			var legacyTraceID string
+			if traceID.High == 0 {
+				legacyTraceID = fmt.Sprintf("%x", traceID.Low)
+			} else {
+				legacyTraceID = fmt.Sprintf("%x%016x", traceID.High, traceID.Low)
+			}
+			query := elastic.NewBoolQuery().
+				Should( elastic.NewTermQuery(traceIDField, traceID.String()).Boost(2), elastic.NewTermQuery(traceIDField, legacyTraceID))
```

Comment:
I am not really sure what goes on in this query, but is there any reason to not simply add the legacy ID to `searchRequests`?

Reply:
It's basically or statement and the new ID has a higher priority https://www.elastic.co/guide/en/elasticsearch/reference/6.6//query-dsl-term-query.html

Reply:
I would rather do this than running a separate query. It might be more effective.